Sources/SwiftGRPCNIO/ClientCalls/BaseClientCall.swift

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+import NIO
+import NIOHTTP1
+import NIOHTTP2
+import SwiftProtobuf
+
+public class BaseClientCall<RequestMessage: Message, ResponseMessage: Message> {
+  /// The underlying `GRPCClient` providing the HTTP/2 channel and multiplexer.
+  internal let client: GRPCClient
```

Comment:
I prefer being explicit but happy to change it if you want?

I kept them `internal` because a lot of these functions should only be called when we're setting up the call: e.g. if you return a `*Call` to the user (which at the moment sends at least the request head), it doesn't make sense for the user to then have the option to e.g. call `createStreamChannel`.

We could always have a more distinct split between the call setup and the object returned to the user if we need to.

Reply:
Let's keep these `internal` until someone actually has a reason to expose them.